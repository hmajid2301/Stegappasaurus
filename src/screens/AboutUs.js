import PropTypes from 'prop-types';
import React, { Component } from 'react';
import { View, Text } from 'react-native';
import { Icon } from 'react-native-elements';

import CustomHeader from '../components/CustomHeader';
import colors from '../components/Common/colors';


export default class AboutUs extends Component {
  static navigationOptions = {
    drawerLabel: 'About Us',
    drawerIcon: ({ tintColor }) => (
      <Icon name='info' type='simple-line-icons' color={tintColor} />
    ),
  };

  static propTypes = {
    navigation: PropTypes.shape({
      navigate: PropTypes.func.isRequired,
    }).isRequired,
  };

  render() {
    return (
      <View>
        <CustomHeader navigation={this.props.navigation} color={colors.quaternaryColor}/>
        <Text> About Us </Text>
      </View>
    );
  }
}
