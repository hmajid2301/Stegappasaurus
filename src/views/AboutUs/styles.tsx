import { StyleSheet } from "react-native";

import { fonts } from "~/modules";

const styles = StyleSheet.create({
  container: {
    flex: 1,
    flexDirection: "column"
  },

  textContainer: {
    alignItems: "center",
    paddingVertical: 20
  },

  about: {
    fontFamily: fonts.bodyLight,
    paddingHorizontal: 20
  }
});

export default styles;
