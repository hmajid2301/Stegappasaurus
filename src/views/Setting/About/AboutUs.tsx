import React, {useContext} from 'react';
import {ScrollView, View} from 'react-native';
import styled from 'styled-components/native';

import AboutList from '~/components/AboutList/AboutList';
import {bodyLight} from '~/constants/fonts';
import {ThemeColors} from '~/constants/types';
import {about} from '~/data';
import {ThemeContext} from '~/providers/ThemeContext';

const AboutUs = () => {
  const {background, color} = useContext(ThemeContext).theme;

  return (
    <AboutContainer background={background}>
      <ScrollView>
        <TextContainer>
          <AboutText color={color}>
            Stegappasaurus is a free mobile application fully open source, built
            using React Native. This application uses steganography algorithms
            to hide your (text) data within images. This project was originally
            created as third year project for University. However this version
            is a complete rewrite of the application.
          </AboutText>
        </TextContainer>

        <View>
          <AboutList items={about(color)} />
        </View>
      </ScrollView>
    </AboutContainer>
  );
};

const AboutContainer = styled.View<{background: ThemeColors}>`
  background: ${props => props.background};
  flex: 1;
  flex-direction: column;
`;

const TextContainer = styled.View`
  align-items: center;
  padding: 20px 0;
`;

const AboutText = styled.Text<{color: ThemeColors}>`
  color: ${props => props.color};
  font-family: ${bodyLight};
  padding: 0 20px;
`;

export default AboutUs;
