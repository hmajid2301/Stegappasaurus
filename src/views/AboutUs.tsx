import React from "react";
import { ScrollView, Text, View } from "react-native";
import { NavigationScreenProp } from "react-navigation";

import { ITheme } from "@types";
import AboutList from "~/components/AboutList";
import AppHeader from "~/components/AppHeader";
import { about } from "~/data";
import styles from "~/views/AboutUs/styles";

interface IProps {
  navigation: NavigationScreenProp<any, any>;
  screenProps: {
    theme: ITheme;
  };
}

export default class AboutUs extends React.Component<IProps, {}> {
  public render() {
    const { theme } = this.props.screenProps;

    return (
      <View style={[styles.container, { backgroundColor: theme.background }]}>
        <ScrollView>
          <AppHeader navigation={this.props.navigation} theme={theme} />

          <View style={styles.textContainer}>
            <Text style={[styles.about, { color: theme.color }]}>
              This project was developed by me as a way to learn React Native.
              It is a complete rewrite of the my university project, which was
              written in Ionic (Apache Cordova). This mobile application allow
              you, the user, to encode image with text using steganography
              algorithms.
            </Text>
          </View>

          <View>
            <AboutList
              items={about}
              backgroundColor={theme.background}
              color={theme.color}
            />
          </View>
        </ScrollView>
      </View>
    );
  }
}
