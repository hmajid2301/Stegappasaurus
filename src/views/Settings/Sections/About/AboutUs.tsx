import React from 'react';
import {ScrollView, Text, View} from 'react-native';

import AboutList from '~/components/AboutList';
import {ThemeColors} from '~/constants/types';
import {about} from '~/data';
import styles from './AboutUs/styles';

interface IProps {
  background: ThemeColors;
  color: ThemeColors;
}

const AboutUs = (props: IProps) => (
  <View style={[styles.container, {backgroundColor: props.background}]}>
    <ScrollView>
      <View style={styles.textContainer}>
        <Text style={[styles.about, {color: props.color}]}>
          Stegappasaurus is a free mobile application fully open source, built
          using React Native. This application uses steganography algorithms to
          hide your (text) data within images. This project was originally
          created as third year project for University. However this version is
          a complete rewrite of the application.
        </Text>
      </View>

      <View>
        <AboutList
          items={about}
          backgroundColor={props.background}
          color={props.color}
        />
      </View>
    </ScrollView>
  </View>
);

export default AboutUs;
