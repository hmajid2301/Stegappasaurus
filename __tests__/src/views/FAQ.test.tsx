import { shallow } from "enzyme";
import React from "react";

import AboutUs from "~/views/AboutUs";
describe("AboutUs: Match Snapshots", () => {
  test("1", () => {
    const component = shallow(
      <AboutUs
        navigation={jest.fn() as any}
        screenProps={{
          theme: {
            color: "#17212D",
            background: "#FFF",
            isDark: false
          }
        }}
      />
    );
    expect(component).toMatchSnapshot();
  });

  test("2", () => {
    const component = shallow(
      <AboutUs
        navigation={jest.fn() as any}
        screenProps={{
          theme: {
            background: "#17212D",
            color: "#FFF",
            isDark: true
          }
        }}
      />
    );
    expect(component).toMatchSnapshot();
  });
});
