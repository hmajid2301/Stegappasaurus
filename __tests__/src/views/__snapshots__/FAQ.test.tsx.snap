// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AboutUs: Match Snapshots 1 1`] = `
<View
  style={
    Array [
      Object {
        "flex": 1,
        "flexDirection": "column",
      },
      Object {
        "backgroundColor": "#FFF",
      },
    ]
  }
>
  <ScrollViewMock>
    <AppHeader
      navigation={[MockFunction]}
      primaryColor="#009CFF"
      theme={
        Object {
          "background": "#FFF",
          "color": "#17212D",
          "isDark": false,
        }
      }
    />
    <View
      style={
        Object {
          "alignItems": "center",
          "paddingVertical": 20,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "fontFamily": "Roboto-Light",
              "paddingHorizontal": 20,
            },
            Object {
              "color": "#17212D",
            },
          ]
        }
      >
        This project was developed by me as a way to learn React Native. It is a complete rewrite of the my university project, which was written in Ionic (Apache Cordova). This mobile application allow you, the user, to encode image with text using steganography algorithms.
      </Text>
    </View>
    <View>
      <AboutList
        backgroundColor="#FFF"
        color="#17212D"
        items={
          Array [
            Object {
              "icon": Object {
                "color": "#009CFF",
                "name": "versions",
                "type": "octicon",
              },
<<<<<<< HEAD
<<<<<<< HEAD
<<<<<<< HEAD
<<<<<<< HEAD
              "title": "Version 1.0.0-beta.6",
=======
              "title": "Version 1.0.0-beta.4",
>>>>>>> 8111df8... Added extra logging
=======
              "title": "Version 1.0.0-beta.5",
>>>>>>> cbb2597... :bug: Fixed Notch in Android
=======
=======
>>>>>>> 72da32f... :bug: Fixed Notch in Android
              "title": "Version 1.0.0-beta.5",
=======
              "title": "Version 1.0.0-beta.4",
>>>>>>> 7a75f16... Added extra logging
<<<<<<< HEAD
>>>>>>> f94da09... Added extra logging
=======
=======
              "title": "Version 1.0.0-beta.5",
>>>>>>> 172e2a2... :bug: Fixed Notch in Android
>>>>>>> 72da32f... :bug: Fixed Notch in Android
              "url": "https://github.com/hmajid2301/Stegappasaurus",
            },
            Object {
              "icon": Object {
                "color": "#F50AE1",
                "name": "code-fork",
                "type": "font-awesome",
              },
              "title": "Fork this Project on GitHub",
              "url": "https://github.com/hmajid2301/Stegappasaurus",
            },
            Object {
              "icon": Object {
                "color": "#FF9501",
                "name": "gitlab",
                "type": "font-awesome",
              },
              "title": "Personal GitLab",
              "url": "https://gitlab.com/hmajid2301",
            },
            Object {
              "icon": Object {
                "color": "#4CDA64",
                "name": "web",
                "type": "material-community",
              },
              "title": "Personal Website",
              "url": "https://hmajid2301.github.io",
            },
            Object {
              "function_to_call": "store",
              "icon": Object {
                "color": "#FD3C2D",
                "name": "rate-review",
                "type": "material",
              },
              "title": "Rate the App",
            },
          ]
        }
      />
    </View>
  </ScrollViewMock>
</View>
`;

exports[`AboutUs: Match Snapshots 2 1`] = `
<View
  style={
    Array [
      Object {
        "flex": 1,
        "flexDirection": "column",
      },
      Object {
        "backgroundColor": "#17212D",
      },
    ]
  }
>
  <ScrollViewMock>
    <AppHeader
      navigation={[MockFunction]}
      primaryColor="#009CFF"
      theme={
        Object {
          "background": "#17212D",
          "color": "#FFF",
          "isDark": true,
        }
      }
    />
    <View
      style={
        Object {
          "alignItems": "center",
          "paddingVertical": 20,
        }
      }
    >
      <Text
        style={
          Array [
            Object {
              "fontFamily": "Roboto-Light",
              "paddingHorizontal": 20,
            },
            Object {
              "color": "#FFF",
            },
          ]
        }
      >
        This project was developed by me as a way to learn React Native. It is a complete rewrite of the my university project, which was written in Ionic (Apache Cordova). This mobile application allow you, the user, to encode image with text using steganography algorithms.
      </Text>
    </View>
    <View>
      <AboutList
        backgroundColor="#17212D"
        color="#FFF"
        items={
          Array [
            Object {
              "icon": Object {
                "color": "#009CFF",
                "name": "versions",
                "type": "octicon",
              },
<<<<<<< HEAD
<<<<<<< HEAD
<<<<<<< HEAD
<<<<<<< HEAD
              "title": "Version 1.0.0-beta.6",
=======
              "title": "Version 1.0.0-beta.4",
>>>>>>> 8111df8... Added extra logging
=======
              "title": "Version 1.0.0-beta.5",
>>>>>>> cbb2597... :bug: Fixed Notch in Android
=======
=======
>>>>>>> 72da32f... :bug: Fixed Notch in Android
              "title": "Version 1.0.0-beta.5",
=======
              "title": "Version 1.0.0-beta.4",
>>>>>>> 7a75f16... Added extra logging
<<<<<<< HEAD
>>>>>>> f94da09... Added extra logging
=======
=======
              "title": "Version 1.0.0-beta.5",
>>>>>>> 172e2a2... :bug: Fixed Notch in Android
>>>>>>> 72da32f... :bug: Fixed Notch in Android
              "url": "https://github.com/hmajid2301/Stegappasaurus",
            },
            Object {
              "icon": Object {
                "color": "#F50AE1",
                "name": "code-fork",
                "type": "font-awesome",
              },
              "title": "Fork this Project on GitHub",
              "url": "https://github.com/hmajid2301/Stegappasaurus",
            },
            Object {
              "icon": Object {
                "color": "#FF9501",
                "name": "gitlab",
                "type": "font-awesome",
              },
              "title": "Personal GitLab",
              "url": "https://gitlab.com/hmajid2301",
            },
            Object {
              "icon": Object {
                "color": "#4CDA64",
                "name": "web",
                "type": "material-community",
              },
              "title": "Personal Website",
              "url": "https://hmajid2301.github.io",
            },
            Object {
              "function_to_call": "store",
              "icon": Object {
                "color": "#FD3C2D",
                "name": "rate-review",
                "type": "material",
              },
              "title": "Rate the App",
            },
          ]
        }
      />
    </View>
  </ScrollViewMock>
</View>
`;
